drv/sidecar-front-io/src/leds.rs

```diff
@@ -0,0 +1,320 @@
+    },
+    // Port 14
+    LedLocation {
+        controller: LedController::Right,
+        output: 15,
+    },
+    // Port 15
+    LedLocation {
+        controller: LedController::Right,
+        output: 13,
+    },
+    // Port 16
+    LedLocation {
+        controller: LedController::Left,
+        output: 3,
+    },
+    // Port 17
+    LedLocation {
+        controller: LedController::Left,
+        output: 1,
+    },
+    // Port 18
+    LedLocation {
+        controller: LedController::Left,
+        output: 7,
+    },
+    // Port 19
+    LedLocation {
+        controller: LedController::Left,
+        output: 5,
+    },
+    // Port 20
+    LedLocation {
+        controller: LedController::Left,
+        output: 9,
+    },
+    // Port 21
+    LedLocation {
+        controller: LedController::Left,
+        output: 11,
+    },
+    // Port 22
+    LedLocation {
+        controller: LedController::Left,
+        output: 14,
+    },
+    // Port 23
+    LedLocation {
+        controller: LedController::Left,
+        output: 12,
+    },
+    // Port 24
+    LedLocation {
+        controller: LedController::Right,
+        output: 3,
+    },
+    // Port 25
+    LedLocation {
+        controller: LedController::Right,
+        output: 1,
+    },
+    // Port 26
+    LedLocation {
+        controller: LedController::Right,
+        output: 7,
+    },
+    // Port 27
+    LedLocation {
+        controller: LedController::Right,
+        output: 5,
+    },
+    // Port 28
+    LedLocation {
+        controller: LedController::Right,
+        output: 9,
+    },
+    // Port 29
+    LedLocation {
+        controller: LedController::Right,
+        output: 11,
+    },
+    // Port 30
+    LedLocation {
+        controller: LedController::Right,
+        output: 14,
+    },
+    // Port 31
+    LedLocation {
+        controller: LedController::Right,
+        output: 12,
+    },
+    // System
+    LedLocation {
+        controller: LedController::Left,
+        output: 23,
+    },
+];
+
+impl Leds {
+    pub fn new(
+        left_controller: &I2cDevice,
+        right_controller: &I2cDevice,
+    ) -> Self {
+        Self {
+            controllers: [
+                Pca9956B::new(left_controller),
+                Pca9956B::new(right_controller),
+            ],
+        }
+    }
+
+    /// Set the current to whatever DEFAULT_LED_CURRENT is
+    pub fn initialize_current(&self) -> Result<(), Error> {
+        self.set_current(DEFAULT_LED_CURRENT)?;
+
+        Ok(())
+    }
+
+    /// Set the current to `value`
+    pub fn set_current(&self, value: u8) -> Result<(), Error> {
+        for (_i, controller) in self.controllers.iter().enumerate() {
+            controller.set_iref_all(value)?;
+        }
+
+        Ok(())
+    }
+
+    /// Turns on the System LED to a PWM value of DEFAULT_LED_PWM
+    pub fn turn_on_system_led(&self) -> Result<(), Error> {
+        self.controllers[LED_MAP[SYSTEM_LED_IDX].controller as usize]
```

Comment:
Instead of looking it up repeatedly, I'd do
```rust
        const SYSTEM_LED: LedLocation = LED_MAP[SYSTEM_LED_IDX];
        self.controllers[SYSTEM_LED.controller as usize]
            .set_a_led_pwm(SYSTEM_LED.output, DEFAULT_LED_PWM)?;
```
(requires `#[derive(Clone)]` on `LedLocation`)

This isn't any more or less efficient, but is slightly more readable IMHO